### PR TITLE
Image model update

### LINKED ISSUE #35: refactor Image model
Remove `type` property.
Change `url` => `data`

because are going to store the image in the db as base64 data

### PR DESCRIPTION
**Pull request by theo333**
Closes #35 

Removed type property.
Changed `url` => `data`

because are going to store the image in the db as base64 data

Also changed tabs => spaces for seed file.  Fewer errors == good

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -7,4 +7,6 @@ app.use(express.static(path.join(__dirname, '../dist')));
 
 app.use(morgan('dev'));
 
+app.use('/api', require('./routes/api'));
```
Comment (leafoflegend) [inline server/app.js]: Why is the `require` inline?

Comment (leafoflegend) [reply server/app.js]: Most linting setups should scream at this.

File: server/db/models/Image.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+// storing images as base64 data
+
 module.exports = conn.define('image', {
   // from associations: challengeId, userchallengeId
-  type: {
-    type: Sequelize.ENUM('challenge', 'userchallenge'),
-  },
-  url: {
-    type: Sequelize.STRING,
+  data: {
```
Comment (leafoflegend) [inline server/db/models/Image.js]: Have you contemplated storing the data as a binary blob? The processing time of converting base64 is more expensive than memory.

File: server/db/seed.js [hunk 3]
```diff
@@ -19,45 +19,45 @@ const challengesSeed = [
...
-		<html>
-		<head>
-			<title>Circle</title>
-		</head>
-		<body>
-			<div id='circle'></div>
-		</body>
-		</html>`,
+    <html>
+    <head>
```
Comment (leafoflegend) [inline server/db/seed.js]: Indentation.

File: server/routes/api/challenges.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// get all challenges
+router.get('/', (req, res, next) => {
+  Challenge.findAll()
+    .then(challenges => res.send(challenges))
+    .catch(next);
+});
+
+// create a single challenge
+router.post('/', (req, res, next) => {
+  Challenge.create(req.body)
```
Comment (leafoflegend) [inline server/routes/api/challenges.js]: Trusting `req.body` quite a bit here eh?

File: server/routes/api/challenges.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+// get a single challenge
+router.get('/:id', (req, res, next) => {
+  Challenge.findByPk(req.params.id)
+    .then(challenge => res.send(challenge))
+    .catch(next);
+});
+
+// update a single challenge
+router.put('/:id', (req, res, next) => {
+  Challenge.findByPk(req.params.id)
```
Comment (leafoflegend) [inline server/routes/api/challenges.js]: What if id is `NaN`?

File: server/routes/api/index.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router();
+
+router.use('/challenges', require('./challenges'));
```
Comment (leafoflegend) [inline server/routes/api/index.js]: Again - why are we inlineing requiring modules?

File: server/routes/api/solutions.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+const router = require('express').Router();
```
Comment (leafoflegend) [inline server/routes/api/solutions.js]: File has similar issues as previously mentioned.

File: server/routes/api/challenges.js (hunk 5 above)
Comment (hgranthorner) [reply server/routes/api/challenges.js]: Doesn't Sequelize coerce to number in this situation? http://docs.sequelizejs.com/class/lib/model.js~Model.html

Comment (leafoflegend) [reply server/routes/api/challenges.js]: `curl localhost:3000/api/challenges/am_i_coercable?`

Comment (leafoflegend) [reply server/routes/api/challenges.js]: Its not about an error, its about returning the appropriate response. If you dont receive a number, the database should never hear about it.

File: server/app.js (hunk 1 above)
Comment (theo333) [reply server/app.js]: Fixed.  Linter didn't even whisper.  Looking to add a rule, but not sure what to add when looking at rules here:  [https://eslint.org/docs/rules/](url)

File: server/db/models/Image.js (hunk 2 above)
Comment (theo333) [reply server/db/models/Image.js]: Have never worked with them.  Will look them up.  Thanks.

File: server/db/seed.js (hunk 3 above)
Comment (theo333) [reply server/db/seed.js]: Fixed.

File: server/routes/api/index.js (hunk 6 above)
Comment (theo333) [reply server/routes/api/index.js]: Fixed

File: server/routes/api/challenges.js (hunk 4 above)
Comment (theo333) [reply server/routes/api/challenges.js]: Changed it to this:
`// create a single challenge
router.post('/', (req, res, next) => {
  const { name, description, difficulty } = req.body;
  Challenge.create({
    name,
    description,
    difficulty
  })
    .then(challenge => res.send(challenge))
    .catch(next);
});`

File: server/db/index.js [hunk 8]
```diff
@@ -18,12 +18,12 @@ const syncAndSeed = () => {
         Promise.all(challengesSeed.map(chal => Challenge.create(chal))),
         Promise.all(userchallengeSeed.map(chal => Userchallenge.create(chal))),
         Promise.all(solutionsSeed.map(sol => Solution.create(sol))),
-        Promise.all(imagesSeed.map(img => Image.create(img))),
+        Promise.all(
+          imagesSeed.map(img => Image.create({ connector: img.connector, data: img.data })),
+        ),
       ]);
     })
     .then(([users, challenges, userchallenges, solutions, images]) => {
```
Comment (chem2016) [inline server/db/index.js]: users and challenges are not used in the code below, how about in stead of using hard code number 3, 2 on line 33, we use sth like users[Math.floor(Math.random() * 10)].id and challenges[Math.floor(Math.random() * 10)].id

**Pull request merged**